pkg/issuer/acme/acme.go

```diff
@@ -73,31 +89,53 @@ func New(issuer v1alpha1.GenericIssuer,
 		return nil, fmt.Errorf("resource namespace cannot be empty")
 	}
 
-	return &Acme{
-		issuer:                   issuer,
-		client:                   client,
-		cmClient:                 cmClient,
-		recorder:                 recorder,
-		secretsLister:            secretsLister,
+	a := &Acme{
+		issuer:         issuer,
+		client:         client,
+		cmClient:       cmClient,
+		recorder:       recorder,
+		secretsLister:  secretsLister,
+		podsLister:     podsLister,
+		servicesLister: servicesLister,
+		ingressLister:  ingressLister,
+
 		dnsSolver:                dns.NewSolver(issuer, client, secretsLister, resourceNamespace),
-		httpSolver:               http.NewSolver(issuer, client, secretsLister, acmeHTTP01SolverImage),
+		httpSolver:               http.NewSolver(issuer, client, podsLister, servicesLister, ingressLister, acmeHTTP01SolverImage),
 		issuerResourcesNamespace: resourceNamespace,
-	}, nil
+	}
+	a.acmeClient = a.acmeClientImpl
+	return a, nil
+}
+
+var timeout = time.Duration(5 * time.Second)
+
+func dialTimeout(ctx context.Context, network, addr string) (net.Conn, error) {
+	return net.DialTimeout(network, addr, timeout)
```

Comment:
How come we're not using / wrapping the supplied context?
Won't this prevent the caller being able to cancel the context?